common/lib/util/defaults.js

```diff
@@ -116,16 +127,22 @@ Defaults.normaliseOptions = function(options) {
 	if(!('queueMessages' in options))
 		options.queueMessages = true;
 
-	var production = false;
 	if(options.restHost) {
 		options.realtimeHost = options.realtimeHost || options.restHost;
+		options.fallbackHosts = options.fallbackHostsUseDefault && !options.port && !options.tlsPort ? Defaults.FALLBACK_HOSTS : options.fallbackHosts;
```

Comment:
returns hosts based on options.fallbackHostsUseDefault. Prioroty is given to options.fallbackHostsUseDefault to set fallback hosts

Reply:
Yes you are correct this should prioritise `options.fallbackHosts`

Reply:
@lmars to be clear, is this a bug that we need to fix in this library? If so, are you aware whether an issue or PR has been created or is WIP yet?

Reply:
After re-reading the spec, as per [`TO3k7`](https://www.ably.io/documentation/client-lib-development-guide/features#TO3k7) `fallbackHosts` and `fallbackHostsUseDefault` are mutually exclusive, so there should be no sense of prioritising one over the other, and there should be an explicit check that they are not both set (e.g. like we do in ably-ruby [here](https://github.com/ably/ably-ruby/blob/main/lib/ably/rest/client.rb#L177-L179)).

Reply:
Quite a good example of where a factory type approach for client options would iron these sorts of issues out without the need for runtime exceptions at the point of initialisation of the operational Ably instance! If we can find the opportunity to engineer ClientOptions-v2 then that's going to be my preferred approach, for sure. 🤔

Reply:
Yes, you can throw an error if both are set : )

Reply:
This is addressed in https://github.com/ably/ably-js/pull/689.